src/NHibernate/Linq/Assignments.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using NHibernate.Linq.Visitors;
+
+namespace NHibernate.Linq
+{
+	/// <summary>
+	/// Class to hold assignments used in updates and inserts.
+	/// </summary>
+	/// <typeparam name="TSource">The type of the entity source of the insert or to update.</typeparam>
+	/// <typeparam name="TTarget">The type of the entity to insert or to update.</typeparam>
+	public class Assignments<TSource, TTarget>
+	{
+		private readonly Dictionary<string, Expression> _assignments = new Dictionary<string, Expression>();
+
+		internal IReadOnlyDictionary<string, Expression> List => _assignments;
+
+		/// <summary>
+		/// Sets the specified property.
+		/// </summary>
+		/// <typeparam name="TProp">The type of the property.</typeparam>
+		/// <param name="property">The property.</param>
+		/// <param name="expression">The expression that should be assigned to the property.</param>
+		/// <returns>The current assignments list.</returns>
+		public Assignments<TSource, TTarget> Set<TProp>(Expression<Func<TTarget, TProp>> property, Expression<Func<TSource, TProp>> expression)
```

Comment:
We would need an additional method which accepts `Expression<Func<TProp>> expression` to be able to write following:

```
.Set(x=>x.CreatedAt, () => SqlFunctions.GetDate() /* DateTime.Now in future */)
```